gum/backend-darwin/gumprocess-darwin.c

```diff
@@ -169,11 +169,85 @@ struct _DyldImageInfo64
   guint64 image_file_mod_date;
 };
 
-#ifndef PROC_SETPC_NONE
-extern int proc_regionfilename (int pid, uint64_t address, void * buffer,
-    uint32_t buffersize);
+#ifndef PROC_INFO_CALL_PIDINFO
+
+#define PROC_INFO_CALL_PIDINFO 0x2
+#define PROC_PIDREGIONPATHINFO 8
+
+struct vinfo_stat
+{
+  uint32_t vst_dev;
+  uint16_t vst_mode;
+  uint16_t vst_nlink;
+  uint64_t vst_ino;
+  uid_t vst_uid;
+  gid_t vst_gid;
+  int64_t vst_atime;
+  int64_t vst_atimensec;
+  int64_t vst_mtime;
+  int64_t vst_mtimensec;
+  int64_t vst_ctime;
+  int64_t vst_ctimensec;
+  int64_t vst_birthtime;
+  int64_t vst_birthtimensec;
+  off_t vst_size;
+  int64_t vst_blocks;
+  int32_t vst_blksize;
+  uint32_t vst_flags;
+  uint32_t vst_gen;
+  uint32_t vst_rdev;
+  int64_t vst_qspare[2];
+};
+
+struct vnode_info
+{
+  struct vinfo_stat vi_stat;
+  int vi_type;
+  int vi_pad;
+  fsid_t vi_fsid;
+};
+
+struct vnode_info_path
+{
+  struct vnode_info vip_vi;
+  char vip_path[MAXPATHLEN];
+};
+
+struct proc_regioninfo
+{
+  uint32_t pri_protection;
+  uint32_t pri_max_protection;
+  uint32_t pri_inheritance;
+  uint32_t pri_flags;
+  uint64_t pri_offset;
+  uint32_t pri_behavior;
+  uint32_t pri_user_wired_count;
+  uint32_t pri_user_tag;
+  uint32_t pri_pages_resident;
+  uint32_t pri_pages_shared_now_private;
+  uint32_t pri_pages_swapped_out;
+  uint32_t pri_pages_dirtied;
+  uint32_t pri_ref_count;
+  uint32_t pri_shadow_depth;
+  uint32_t pri_share_mode;
+  uint32_t pri_private_pages_resident;
+  uint32_t pri_shared_pages_resident;
+  uint32_t pri_obj_id;
+  uint32_t pri_depth;
+  uint64_t pri_address;
+  uint64_t pri_size;
+};
+
+struct proc_regionwithpathinfo
+{
+  struct proc_regioninfo prp_prinfo;
+  struct vnode_info_path prp_vip;
+};
 #endif
```

Comment:
Let's add a blank line above `#endif` to balance it with the blank line after the opening `#ifndef`.